src/uipath/_cli/_evals/_runtime.py

```diff
@@ -86,30 +99,52 @@ def __init__(self, context: UiPathEvalContext, factory: UiPathRuntimeFactory[T,
     def from_eval_context(
         cls,
         context: UiPathEvalContext,
-        factory: UiPathRuntimeFactory[T, C],
+        factory: UiPathRuntimeFactory[T, C]
     ) -> "UiPathEvalRuntime[T, C]":
         return cls(context, factory)
 
     async def execute(self) -> Optional[UiPathRuntimeResult]:
-        """Evaluation logic. Can spawn other runtimes through the factory."""
         if self.context.eval_set is None:
             raise ValueError("eval_set must be provided for evaluation runs")
 
+        event_bus = await get_event_bus()
+
         evaluation_set = EvalHelpers.load_eval_set(
             self.context.eval_set, self.context.eval_ids
         )
         evaluators = self._load_evaluators(evaluation_set)
+
+        await event_bus.publish(
+            EvaluationEvents.CREATE_EVAL_SET_RUN,
+            CreateEvalSetRunEvent(
+                eval_set_id=evaluation_set.id,
+                agent_snapshot=self._extract_agent_snapshot(),
```

Comment:
how about we send entrypoint/input_schema/output_schema in the event (and create the SWAgentSnapshot model in the subscriber?